### PR TITLE
posix: unistd: support for confstr()

### LINKED ISSUE #66866: posix: implement `confstr()`
This is part of the See [POSIX Roadmap for LTSv3](https://github.com/zephyrproject-rtos/zephyr/issues/51211) (RFC #51211).

`confstr()` is required as part of `POSIX_SINGLE_PROCESS` [Subprofiling Option Group](https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_subprofiles.html).

For more information, please refer to https://pubs.opengroup.org/onlinepubs/9699919799/functions/confstr.html

### PR DESCRIPTION
**Pull request by cfriedt**
Support getting and setting POSIX configuration variables.

`confstr()` is required by the `POSIX_SINGLE_PROCESS` Option Group as detailed in Section E.1 of IEEE-1003.1-2017.

The POSIX_SINGLE_PROCESS Option Group is required for PSE51, PSE52, PSE53, and PSE54 conformance, and is otherwise mandatory for any POSIX conforming system as per Section A.2.1.3 of IEEE-1003-1.2017.

With this, we have complete support for the `POSIX_SINGLE_PROCESS` Option Group.

Fixes #66866

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]

File: include/zephyr/posix/sys/confstr.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+_CS_POSIX_V6_ILP32_OFFBIG_LIBS,
+_CS_POSIX_V6_LP64_OFF64_CFLAGS,
+_CS_POSIX_V6_LP64_OFF64_LDFLAGS,
+_CS_POSIX_V6_LP64_OFF64_LIBS,
+_CS_POSIX_V6_LPBIG_OFFBIG_CFLAGS,
+_CS_POSIX_V6_LPBIG_OFFBIG_LDFLAGS,
+_CS_POSIX_V6_LPBIG_OFFBIG_LIBS,
+_CS_POSIX_V6_WIDTH_RESTRICTED_ENVS,
+_CS_V6_ENV,
+};
```
Comment (ycsin) [inline include/zephyr/posix/sys/confstr.h:45]:
```suggestion
enum {
	_CS_PATH,
	_CS_POSIX_V7_ILP32_OFF32_CFLAGS,
	_CS_POSIX_V7_ILP32_OFF32_LDFLAGS,
	_CS_POSIX_V7_ILP32_OFF32_LIBS,
	_CS_POSIX_V7_ILP32_OFFBIG_CFLAGS,
	_CS_POSIX_V7_ILP32_OFFBIG_LDFLAGS,
	_CS_POSIX_V7_ILP32_OFFBIG_LIBS,
	_CS_POSIX_V7_LP64_OFF64_CFLAGS,
	_CS_POSIX_V7_LP64_OFF64_LDFLAGS,
	_CS_POSIX_V7_LP64_OFF64_LIBS,
	_CS_POSIX_V7_LPBIG_OFFBIG_CFLAGS,
	_CS_POSIX_V7_LPBIG_OFFBIG_LDFLAGS,
	_CS_POSIX_V7_LPBIG_OFFBIG_LIBS,
	_CS_POSIX_V7_THREADS_CFLAGS,
	_CS_POSIX_V7_THREADS_LDFLAGS,
	_CS_POSIX_V7_WIDTH_RESTRICTED_ENVS,
	_CS_V7_ENV,
	_CS_POSIX_V6_ILP32_OFF32_CFLAGS,
	_CS_POSIX_V6_ILP32_OFF32_LDFLAGS,
	_CS_POSIX_V6_ILP32_OFF32_LIBS,
	_CS_POSIX_V6_ILP32_OFFBIG_CFLAGS,
	_CS_POSIX_V6_ILP32_OFFBIG_LDFLAGS,
	_CS_POSIX_V6_ILP32_OFFBIG_LIBS,
	_CS_POSIX_V6_LP64_OFF64_CFLAGS,
	_CS_POSIX_V6_LP64_OFF64_LDFLAGS,
	_CS_POSIX_V6_LP64_OFF64_LIBS,
	_CS_POSIX_V6_LPBIG_OFFBIG_CFLAGS,
	_CS_POSIX_V6_LPBIG_OFFBIG_LDFLAGS,
	_CS_POSIX_V6_LPBIG_OFFBIG_LIBS,
	_CS_POSIX_V6_WIDTH_RESTRICTED_ENVS,
	_CS_V6_ENV,
};
```

**Review by ycsin — DISMISSED** [review]
LGTM

File: include/zephyr/posix/sys/confstr.h (hunk 1 above)
Comment (cfriedt) [reply include/zephyr/posix/sys/confstr.h:45]: Fixed!

**Comment by cfriedt** [comment]
* manually inserted a tab. For some reason, autoformat is not working for me 😢 

@jukkar @ycsin - please re-approve, if you don't mind

**Review by ycsin — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**